src/TypeSpec.Extension/Emitter.Csharp/src/options.ts

```diff
@@ -33,6 +33,7 @@ export type NetEmitterOptions = {
     branded?: boolean;
     generateTestProject?: boolean;
     "use-model-reader-writer"?: boolean;
+    arm?: boolean;
```

Comment:
could we keep the name aligned with the config in swagger? It is called `azure-arm`.

Reply:
`arm` is defined in sdkContext, which is not in our control.
I think this option is a cmobination of sdkContext and .net generator options.

Reply:
yes, it should be, will update it.

Reply:
Actually this is not needed, because `arm` is already defined in `SdkEmitterOptions`

Reply:
I'm confused how user pass in `arm`? They only way I know is from `tspconfig.yaml`. Do we have other ways?

Reply:
user doesn't need to pass in `arm`, it is from `sdkContext.arm` from tcgc

Reply:
tcgc already knows if the input TSP is `arm`

Reply:
I see, it is assigned after we call listClients.  https://github.com/Azure/typespec-azure/blob/0a5020b2964d7fc565ddc42ad09147b4b22cf3f4/packages/typespec-client-generator-core/src/decorators.ts#L209. So I'm afraid your code only works when there is only one sdkContext instance. Put it another way, you still need Mingzhe's change.

Reply:
BTW, why not we call `isArm(service.type)` ourselves? Why we have to read it from sdkContext?

Reply:
Please check the latest code, we don't have `service` in `emitter.ts`

Reply:
> I see, it is assigned after we call listClients. https://github.com/Azure/typespec-azure/blob/0a5020b2964d7fc565ddc42ad09147b4b22cf3f4/packages/typespec-client-generator-core/src/decorators.ts#L209. So I'm afraid your code only works when there is only one sdkContext instance. Put it another way, you still need Mingzhe's change.

This works as long as we are using the same `sdkContext` to get `sdkContext.arm`, which is the case now.